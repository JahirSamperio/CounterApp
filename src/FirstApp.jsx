import PropTypes from 'prop-types';

export const FirstApp = () => {
    return (
        <>
            <Form 
                // title= "Jahir" 
                age= {30}
                // name= 'Jahir Samperio'
            />
        </>
    )
}

const Form = ({title, age, name}) => {
    return (
        <>
            <h1>{title}</h1>
            <p>{age}</p>
            <p>{name}</p>
        </>
    )
}

Form.propTypes = {
    title: PropTypes.string.isRequired
}

//Valores por defecto a cada componente
//Entran antes de los proptypes
Form.defaultProps = {
    title: 'No hay titulo',
    name: 'Jahir Samperio'
}