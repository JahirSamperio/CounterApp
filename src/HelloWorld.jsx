//Funcion con el JSX
export const App = () => {
    //document.createElement...
    return <h1>Hola mundo</h1>;
}

export const HelloWorld = () => {
  return (
    <h1>HelloWorld</h1>
  )
}
