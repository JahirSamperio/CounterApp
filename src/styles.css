html, body {
    background-color: blueviolet;
    color: white;
    font-family: Helvetica,Arial, sans-serif;
    font-size: 1.3rem;
    padding: 70px;
}

button {
    padding: 5px;
    background-color: rosybrown;
    font-size: 1rem;
    font-weight: bold;
    
}