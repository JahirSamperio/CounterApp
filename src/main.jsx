//Aplicacion de React
import React from 'react';
//Herramienta para renderizarlo
import ReactDOM from 'react-dom/client';
import { HelloWorld } from './HelloWorld';
import { FirstApp } from './FirstApp';
import { CounterApp } from './CounterApp';
import './styles.css'

//Renderizado de la funcion
ReactDOM.createRoot(document.getElementById('root')).render( //El id debe ser igual al que esta en el div del index
    <React.StrictMode>
        <CounterApp value={ 0 }/> 
        {/* <FirstApp title='Hola soy Jahir' /> */}
    </React.StrictMode>
)/*Se escribre el nombre de la funcion*/
