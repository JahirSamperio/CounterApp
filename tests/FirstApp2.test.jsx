import { render, screen } from "@testing-library/react"
import { FirstApp } from "../src/FirstApp"

describe('Pruebas en <FirstApp />', () => { 
    const title = 'Hola soy Jahir'
    test('debe hacer match con el snapshot', () => { 
        const {container} = render( <FirstApp title={title} />)
        expect(container).toMatchSnapshot();
    })

    test('debe de mostrar el mensaje', () => { 
        render(<FirstApp title={title}/>);
        
        //Screen toma el componente renderizado
        // screen.debug() muestra el componente 

        expect(screen.getByText(title)).toBeTruthy();
    })
    
    test('debe de mostrar el titulo en un h1', () => { 
        render(<FirstApp title={title}/>);
        expect(screen.getByRole('heading', {level: 1}).innerHTML).toContain(title);
     })
})