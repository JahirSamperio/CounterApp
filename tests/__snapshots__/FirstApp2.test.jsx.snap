// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> debe hacer match con el snapshot 1`] = `
<div>
  <div
    data-testid="test-title"
  >
    Hola soy Jahir
  </div>
  <p>
    19
  </p>
  <p>
    Jahir Samperio
  </p>
</div>
`;
